### ISSUE TITLE
Open to rustix Linux implementation?

### ISSUE THREAD

**Issue by SUPERCILEX** [m-ou-se/atomic-wait#3]
I'd like to get pure assembly syscalls which rustix will handle. Thoughts?

**Comment by m-ou-se** [comment]
(Optionally) making this crate do syscalls without libc sounds great. (Whether rustix or another crate is the best option for that, I don't know. ^^)

**Comment by SUPERCILEX** [comment]
> Whether rustix or another crate is the best option for that, I don't know

Fair point. I'm biased towards liking rustix (have made contributions), so I think I'll still choose it for the impl and then you can decided whether or not you want that.

**Comment by SUPERCILEX** [comment]
Hmmm, actually I just realized the wait API doesn't accept a timeout/deadline. Is this because of macOS? I couldn't find much info on this and am not well versed in cross platform development. Is this fixable? If so can you point me to the right macOS APIs and I'll look into it.

**Comment by m-ou-se** [comment]
> Hmmm, actually I just realized the wait API doesn't accept a timeout/deadline. Is this because of macOS?

Yes. I opened a new issue to discuss that: https://github.com/m-ou-se/atomic-wait/issues/4
